src/dev/build/tasks/create_cdn_assets_task.ts

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { readFileSync } from 'fs';
+import { access } from 'fs/promises';
+
+import { resolve, dirname } from 'path';
+import { asyncForEach } from '@kbn/std';
+import { Jsonc } from '@kbn/repo-packages';
+
+import del from 'del';
+import globby from 'globby';
+
+import { mkdirp, compressTar, Task, copyAll } from '../lib';
+
+export const CreateCdnAssets: Task = {
+  description: 'Creating CDN assets',
+
+  async run(config, log, build) {
+    const buildSource = build.resolvePath();
+    const buildNum = config.getBuildNumber();
+    const buildVersion = config.getBuildVersion();
+    const assets = config.resolveFromRepo('build', 'cdn-assets');
+    const bundles = resolve(assets, String(buildNum), 'bundles');
+    const plugin = resolve(bundles, 'plugin');
+
+    await del(assets);
+    await mkdirp(assets);
+
+    // Plugins
+    await mkdirp(plugin);
+    const plugins = globby.sync([`${buildSource}/node_modules/@kbn/**/*/kibana.jsonc`]);
+    await asyncForEach(plugins, async (path) => {
+      const manifest = Jsonc.parse(readFileSync(path, 'utf8')) as any;
+      if (manifest?.plugin?.id) {
+        const pluginRoot = resolve(dirname(path));
+        const pluginAssets = resolve(pluginRoot, 'assets');
+        const pluginBundles = resolve(pluginRoot, 'target', 'public');
+        const dest = resolve(plugin, manifest.plugin.id, '1.0.0');
+        try {
+          await access(pluginAssets);
+          await copyAll(pluginAssets, dest);
```

Comment:
needs to be something like
```suggestion
          await copyAll(pluginAssets, resolve(dest, 'assets'));
```

Reply:
I pushed https://github.com/elastic/kibana/pull/169707/commits/01c6fb5cbe5277a959690c1add1429d81fc90471